bin/i18n/resources/pegasus/mobile/sync_in.rb

```diff
@@ -10,9 +10,17 @@ module Pegasus
       module Mobile
         class SyncIn < I18n::Utils::SyncInBase
           def process
-            # TODO: fix `i18n/locales/source/pegasus/mobile.yml` instead of the original file `pegasus/cache/i18n/en-US.yml`
-            I18nScriptUtils.fix_yml_file(ORIGINAL_I18N_FILE_PATH)
-            I18nScriptUtils.copy_file(ORIGINAL_I18N_FILE_PATH, I18N_SOURCE_FILE_PATH)
+            i18n_data = I18nScriptUtils.parse_file(ORIGINAL_I18N_FILE_PATH)
+            if i18n_data && i18n_data['en-US']
+              I18nScriptUtils.sanitize_data_and_write(i18n_data['en-US'], I18N_SOURCE_FILE_PATH)
+            else
+              # Handle the case where i18n_data is nil or does not contain 'en-US'
+              handle_missing_or_invalid_data
+            end
+          end
+
+          def handle_missing_or_invalid_data
+            raise "Error: i18n data is missing or invalid"
```

Comment:
Can we provide more information to identify which locale is missing its i18n data?